src/srv.rs

```diff
@@ -24,7 +24,7 @@ pub(crate) struct ResolvedConfig {
 
 #[derive(Debug, Clone)]
 pub(crate) struct LookupHosts {
-    pub(crate) hosts: Vec<Result<ServerAddress>>,
```

Comment:
This being a `Vec<Result<ServerAddress>>` rather than `Vec<ServerAddress>` with error values either being propagated or ignored depending on context made me stumble a bit when reading this code, so I switched it to the (IMO) more understandable simple value vec with an explicit behavior toggle.